### PR TITLE
feat(profiling): flamechart search ui enhancements

### PR DESCRIPTION
**Pull request by eliashussary**
## Summary

This PR contains some minor UI enhancements to improve the `flamechart` search ux.

~~- debounces the search input as it would previously block the event loop causing the UI to hang~~
- adds search result cycling  similar to `cmd+f` in chrome
~~- highlights searchResults in minimap~~

Closes: https://github.com/getsentry/team-profiling/issues/104

Note: after some exploration, `fuse.search` seems to be the bottleneck causing jank. its problematic with an increasingly large query where fuse needs to take more of a fuzzy approach. attempts to make the fuzzy matching stricter doesn't improve performance and also changes the behaviour of highlihting substrings. 

tldr; either we always debounce, debounce under certain conditions (this can get a bit out of control), or we explore something other than fuse.

Before w/ jank:

https://user-images.githubusercontent.com/7349258/202803171-a2edf865-1f70-4b09-990c-c672d93f4eb2.mov

After - debounced w/o jank + search result cycling:

https://user-images.githubusercontent.com/7349258/202803264-2d053a9e-4a84-40a2-882e-a8411068a24f.mov

### CODE REVIEW

File: static/app/components/profiling/flamegraphSearch.tsx [hunk 1]
```diff
@@ -321,13 +327,45 @@ function FlamegraphSearch({
     <StyledSearchBar
       size="xs"
       placeholder={t('Find Frames')}
-      query={search.query}
+      defaultQuery={search.query}
```
Comment (eliashussary) [inline static/app/components/profiling/flamegraphSearch.tsx:330]: just delegate the input value rendering to the `SearchBar` - app search state can remain in `flamegraphSearch`

File: static/app/components/profiling/flamegraphSearch.tsx [hunk 2]
```diff
@@ -321,13 +327,45 @@ function FlamegraphSearch({
...
+              type="button"
+              size="zero"
+              borderless
+              icon={<IconChevron size="xs" direction="down" />}
+              aria-label={t('Previous')}
+              onClick={onPreviousSearchClick}
+            />
+          </Fragment>
+        )
+      }
```
Comment (eliashussary) [inline static/app/components/profiling/flamegraphSearch.tsx:359]: @getsentry/core-ui  not sure if you would see this becoming a permanent option in the `SearchBar` 
image


for now I've just allowed us to add it via a `trailing` prop

File: static/app/components/profiling/flamegraphZoomViewMinimap.tsx [hunk 3]
```diff
@@ -44,6 +45,7 @@ function FlamegraphZoomViewMinimap({
   >(null);
 
   const dispatch = useDispatchFlamegraphState();
+  const flamegraphSearch = useFlamegraphSearch();
```
Comment (eliashussary) [inline static/app/components/profiling/flamegraphZoomViewMinimap.tsx:48]: @JonasBa ; not sure if you have strong opinions on rendering the highlighted frames in the minimap, but i've added it here

Comment (JonasBa) [reply static/app/components/profiling/flamegraphZoomViewMinimap.tsx:48]: Meh, I wouldnt. It's too small to see tbh

File: static/app/components/profiling/flamegraphSearch.tsx [hunk 4]
```diff
@@ -241,21 +242,26 @@ function FlamegraphSearch({
...
-        type: 'set results',
-        payload: {
-          results: frameSearch(value, allFrames, searchIndex),
-          query: value,
-        },
-      });
-    },
+  const handleChange = useMemo(
+    () =>
+      debounce(value => {
```
Comment (JonasBa) [inline static/app/components/profiling/flamegraphSearch.tsx:247]: I have to be honest, I'm not a fan of debouncing because we end up penalizing the majority of the use cases because of occasional outliers. Do we know why search is slow here? Maybe we can do something there and optimize it, otherwise wdyt about enabling debouncing based on some heuristic like nb frames or something?

**Review by JonasBa — COMMENTED** [review]
Love the UI changes, but I want to avoid debouncing or I'd least make it conditional if possible

File: static/app/components/profiling/flamegraphSearch.tsx (hunk 4 above)
Comment (JonasBa) [reply static/app/components/profiling/flamegraphSearch.tsx:247]: >Every keystroke completely re-renders the page.

I think I noticed somthing similar already and it was due to us updating quert string on each keystroke and codebase that used useLocation

>Besides that, the obvious happens; we take the search results and have to re-render the canvas(s) on every keystroke, iterating over all frames at least twice, once for rects, once for text. etc. etc.

I think this is fine as long as we can do it in time.

One different way of tackling this issue without debouncing is to not allow searching on short queries. For example, if a user types something like a letter `a` into the input as their first char, it will likely match 90% of the frames and we'll end up drawing everything even though results are likely useless. If we for example limit shortest search to 3 char, we increase the specificity of the query at the expense of losing those first few characters which eod I'm not a fan of either.

But all of this is hard to say because we dont really know where the performance penalty comes from. For example, if searching through something like 10k frames is slow, then debouncing might make sense, but maybe has more to do with how we end up rerendering the entire page or very likely a combo of both. In either way, I'd look at improving that part before settling on debouncing or make it conditional for some N of frames.

**Review by JonasBa — APPROVED** [review]
